presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -421,16 +425,23 @@ private Estimate summarizePartitionStatistics(
             checkArgument(hivePartitions.size() == 1, "expected only one hive partition");
         }
 
-        if (unpartitioned) {
-            return ImmutableMap.of(HivePartition.UNPARTITIONED_ID, metastore.getTableStatistics(tableHandle.getSchemaName(), tableHandle.getTableName()));
+        try {
+            if (unpartitioned) {
+                return ImmutableMap.of(HivePartition.UNPARTITIONED_ID, metastore.getTableStatistics(tableHandle.getSchemaName(), tableHandle.getTableName()));
+            }
+            else {
+                Set<String> partitionNames = hivePartitions.stream()
+                        .map(HivePartition::getPartitionId)
+                        .collect(toImmutableSet());
+                return metastore.getPartitionStatistics(tableHandle.getSchemaName(), tableHandle.getTableName(), partitionNames);
+            }
         }
-        else {
-            return metastore.getPartitionStatistics(
-                    tableHandle.getSchemaName(),
-                    tableHandle.getTableName(),
-                    hivePartitions.stream()
-                            .map(HivePartition::getPartitionId)
-                            .collect(toImmutableSet()));
+        catch (PrestoException e) {
+            if (e.getErrorCode().getCode() == HIVE_CORRUPTED_COLUMN_STATISTICS.toErrorCode().getCode() && isIgnoreCorruptedStatistics(session)) {
```

Comment:
Since this is an enum, we can compare it like this (see `CreateTableTask` or any of the tests as an example):
```java
if (e.getErrorCode().equals(HIVE_CORRUPTED_COLUMN_STATISTICS.toErrorCode())) && ...)